tests/generic_config_updater/test_mgmt_interface.py

```diff
@@ -0,0 +1,126 @@
+import ipaddress
+import logging
+import pytest
+
+from tests.common.helpers.assertions import pytest_assert
+from tests.generic_config_updater.gu_utils import apply_patch
+from tests.generic_config_updater.gu_utils import generate_tmpfile, delete_tmpfile
+from tests.generic_config_updater.gu_utils import create_checkpoint, delete_checkpoint, rollback_or_reload
+from tests.route.test_forced_mgmt_route import FORCED_MGMT_ROUTE_PRIORITY, wait_for_file_changed
+
+pytestmark = [
+    pytest.mark.topology('any'),
+]
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope="function")
+def ensure_dut_readiness(duthost):
+    """
+    Setup/teardown fixture for pg headroom update test
+
+    Args:
+        duthost: DUT host object
+    """
+    create_checkpoint(duthost)
+
+    yield
+
+    try:
+        logger.info("Rolled back to original checkpoint")
+        rollback_or_reload(duthost)
+    finally:
+        delete_checkpoint(duthost)
+
+
+def test_forced_mgmt_routes_update(duthost, ensure_dut_readiness):
+    # Get interface and check config generate correct
+    mgmt_interface_keys = duthost.command("sonic-db-cli  CONFIG_DB keys 'MGMT_INTERFACE|eth0|*'")['stdout']
+    logging.debug("mgmt_interface_keys: {}".format(mgmt_interface_keys))
+
+    for interface_key in mgmt_interface_keys.split('\n'):
+        logging.debug("interface_key: {}".format(interface_key))
+        interface_address = interface_key.split('|')[2]
+
+        # Get current forced mgmt routes
+        forced_mgmt_routes_config = duthost.command("sonic-db-cli CONFIG_DB HGET '{}' forced_mgmt_routes@"
+                                                    .format(interface_key))['stdout']
+
+        original_forced_mgmt_routes = []
+        if forced_mgmt_routes_config != "":
+            original_forced_mgmt_routes = forced_mgmt_routes_config.split(",")
+
+        # Prepare new forced mgmt routes
+        test_route = "1::2:3:4/64"
+        ip_type = "-6"
+        if type(ipaddress.ip_network(interface_address, False)) == ipaddress.IPv4Network:
+            test_route = "1.2.3.4/24"
+            ip_type = "-4"
+
+        updated_forced_mgmt_routes = original_forced_mgmt_routes.copy()
+        updated_forced_mgmt_routes.append(test_route)
+        logging.debug("interface address: {}, original_forced_mgmt_routes: {}, updated_forced_mgmt_routes: {}"
+                      .format(interface_address, original_forced_mgmt_routes, updated_forced_mgmt_routes))
+
+        def update_forced_mgmt_route(duthost, routes):
+            # Escape '/' in interface key
+            json_patch = [
+                {
+                    "path": "/MGMT_INTERFACE/eth0|{}/forced_mgmt_routes".format(interface_address.replace("/", "~1"))
```

Comment:
suggest to reuse https://github.com/sonic-net/sonic-mgmt/blob/defc8bcde15abb4b5c128b5fef998ba59aea12a0/tests/generic_config_updater/gu_utils.py#L266

Reply:
Fixed